application/client/src/app/ui/views/toolbar/chart/chart/state.ts

```diff
@@ -0,0 +1,261 @@
+import { ISearchMap } from '@platform/interfaces/interface.rust.api.general';
+import { Instance as Logger } from '@platform/env/logger';
+import { Session } from '@service/session';
+import { Chart } from 'chart.js';
+import { StoredEntity } from '@service/session/dependencies/search/store';
+import { FilterRequest } from '@service/session/dependencies/search/filters/request';
+import { IPositionChange } from '@ui/service/toolbar/chart';
+import { Env, InternalAPI } from '@service/ilc';
+
+export interface IRange {
+    begin: number;
+    end: number;
+}
+
+export class State {
+    private _ilc!: InternalAPI;
+    private _env!: Env;
+    private _session!: Session;
+    private _logger!: Logger;
+    private _labels: string[] = [];
+    private _datasets: any[] = [];
+    private _map: ISearchMap = [];
+    private _detectChanges!: () => void;
+    private _filters: Chart | undefined;
+    private _loading: boolean = false;
+
+    public init(
+        session: Session,
+        logger: Logger,
+        containerWidth: number,
+        detectChanges: () => void,
+    ) {
+        this._session = session;
+        this._logger = logger;
+        this._labels = new Array(containerWidth).fill('');
+        this._detectChanges = detectChanges;
+        this._fetch(containerWidth);
+        this._subscribe(containerWidth);
+    }
+
+    public set ilc(ilc: InternalAPI) {
+        this._ilc = ilc;
+    }
+
+    public set env(env: Env) {
+        this._env = env;
+    }
+
+    public noData(): boolean {
+        let displayFilter: boolean = false;
+        if (
+            this._filters &&
+            this._filters.data.datasets &&
+            this._filters.data.datasets.length > 0
+        ) {
+            displayFilter = true;
+        }
+        return !this._loading && !displayFilter;
+    }
+
+    public get loading(): boolean {
+        return this._loading;
+    }
+
+    public get filters(): Chart | undefined {
+        return this._filters;
+    }
+
+    private _subscribe(containerWidth: number) {
+        this._env.subscriber.register(
+            this._session.search.subjects.get().updated.subscribe(this._onUpdate.bind(this)),
+        );
+        this._env.subscriber.register(
+            this._session.search
+                .store()
+                .filters()
+                .subjects.get()
+                .highlights.subscribe((entity: StoredEntity<FilterRequest>) => {
+                    this._changeColor(entity.definition.uuid, entity.definition.colors.background);
+                }),
+        );
+        this._ilc.channel.ui.sidebar.resize(this._resize(containerWidth)._onUpdate.bind(this));
```

Comment:
```
this._ilc.channel.ui.sidebar.resize(this._resize(containerWidth)._onUpdate.bind(this));
```
it would not work like you want probably.
you are executing `this._resize(containerWidth)` just once during subscribing. And resize never will be triggered on the event.
if you want to resize on event, you shouldn't bind here
```
this._ilc.channel.ui.sidebar.resize(() => {
    // how to get actual value for "containerWidth"?
    this._resize(containerWidth)._onUpdate();
})
```